src/legacy/status_im/communities/e2e.cljs

```diff
@@ -1,35 +1,196 @@
+   {:permission-type constants/community-token-permission-can-view-and-post-channel
+    :token-symbol    stt-symbol
+    :amount          one-stt-in-wei}
+   {:permission-type constants/community-token-permission-can-view-and-post-channel
+    :token-symbol    eth-symbol
+    :amount          one-eth-in-wei}])
+
+(defn channel-names->description
+  [channel-names]
+  (map #(hash-map :name        %1
+                  :permissions [%2])
+       channel-names
+       base-channel-permissions))
+
+(def community-descriptions
+  {:token-gated
+   {:community-name        "Token gated community 25"
+    :community-permissions [{:amount-in-wei   ten-stt-in-wei
+                             :permission-type constants/community-token-permission-become-member
+                             :token-symbol    stt-symbol}]
+    :channel-list          (channel-names->description ["Lions" "Birds" "Trees" "Flowers"])}
+   :snt-admin
+   {:community-name        "SNT Admin Community"
+    :community-permissions [{:amount-in-wei   one-stt-in-wei
+                             :permission-type constants/community-token-permission-become-admin
+                             :token-symbol    stt-symbol}]
+    :channel-list          (channel-names->description ["Sounds" "Colors" "Books" "Sports"])}
+   :admin-and-member
+   {:community-name        "Admin and Member"
+    :community-permissions [{:amount-in-wei   one-stt-in-wei
+                             :permission-type constants/community-token-permission-become-member
+                             :token-symbol    stt-symbol}
+                            {:amount-in-wei   one-eth-in-wei
+                             :permission-type constants/community-token-permission-become-admin
+                             :token-symbol    eth-symbol}]
+    :channel-list          (channel-names->description ["Party" "Birthday" "Travel" "Cars"])}})
+
+
+(defn js-response->community-id
+  [response]
+  (-> response .-communities first .-id))
+
+(defn js-response->channel-id
+  [channel-name response]
+  (->> response
+       .-communities
+       first
+       .-chats
+       js->clj
+       vals
+       (some #(when (= channel-name (get % "name")) (get % "id")))))
+
+(defn response->community-id-fn
+  [callback]
+  (comp callback js-response->community-id))
+
+(defn channel-token-criteria->request
+  [community-id channel-id {:keys [token-symbol amount]}]
+  (fn []
+    (rpc/call
+     {:method      "wakuext_createCommunityTokenPermission2"
+      :js-response true
+      :params      [{:communityID   community-id
+                     :type          constants/community-token-permission-become-member
+                     :chatIds       [channel-id]
+                     :tokenCriteria [((token-criteria token-symbol) amount)]}]})))
+
+(defn channel-token-criteria->requests
+  [community-id channel-id tks]
+  (let [keep-fn (partial channel-token-criteria->request community-id channel-id)]
+    (keep keep-fn tks)))
```

Comment:
> Can we rename tks -> tokens? I know is clear for some, but I've seen we prefer to be explicit.

Sure!

> channel-token-criteria->request is a function that always returns a (fn []). so partial will also return it, so there's no need to use keep, using map would be enough.

Yes, sorry, I had the function return nil before in case we didn't specify the token while implementing it, but removed the check and didn't change to map, well spotted!
I can swap out partial as well, that's no problem.

> I know, it's more verbose, but we can clearly see what the map returns.

That's ok as well

> This function made me think it's a bit unnatural having a lazy seq of functions performing side effects, lazy seqs are consumed in chunks by clojure 🤔

This I am not sure, since the function is pure as it returns a lazy seq of functions (which if called, perform side effects), so even if it's chunked/lazy, it's the caller of the function that decides when the side effect is performed, independently on how/when the functions are generated.

i.e 
`(map identity println)` vs `(mapv identity println)` to make the simplest similar case (the return value is a lazy-seq returning side-effectful functions), here I would think that if you use `mapv`, you have a specific reason to do so (i.e you really want a vector for some reason), while `map` would seem more natural if you didn't care (as in the case above).
That's mostly just for the sake of discussion, to be honest :), I am happy to change it to `mapv`/etc, that's perfectly fine

Reply:
>  it's the caller of the function that decides when the side effect is performed, independently on how/when the functions are generated.

That's true, and that's a really good point :+1:  No need to change anything related to this